components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -922,6 +922,37 @@ public AccessTokenDO getAccessToken(String accessTokenIdentifier, boolean includ
         return dataDO;
     }
 
+    public void updateAccessTokenState(String tokenId, String tokenState,
+                                       String userStoreDomain) throws IdentityOAuth2Exception {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Changing status of access token with id: " + tokenId + " to: " + tokenState +
+                        " userStoreDomain: " + userStoreDomain);
+            }
+
+            String sql = SQLQueries.UPDATE_TOKEN_STATE;
```

Comment:
User store partitioning is deprecated. Should we include then in the newly introduced APIs?

Reply:
Done